src/goto-programs/json_goto_trace.cpp

```diff
@@ -97,8 +97,8 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
-          assert(step.full_lhs_value.is_not_nil());
-          full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
+          if(step.full_lhs_value.is_not_nil())
+            full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
```

Comment:
fixed
the variable seems to be eliminated and therefore `prop_conv.get(expr)` returns `nil_exprt`